test/proposal/encrypter_test.rb

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../spec_helper"
+require "y2storage"
+
+describe Y2Storage::Proposal::Encrypter do
+  using Y2Storage::Refinements::SizeCasts
+
+  subject(:encrypter) { described_class.new }
+
+  describe "#device_for" do
+    let(:volume) { planned_vol(mount_point: "/dev/sda1", type: :ext4, desired: 10.GiB) }
+    let(:plain_device) { instance_double("Storage::BlkDevice") }
+    let(:luks) { instance_double("Storage::Encryption") }
+
+    before do
+      allow(plain_device).to receive(:create_encryption).with(anything).and_return(luks)
+      allow(plain_device).to receive(:name).and_return(volume.mount_point)
+      allow(luks).to receive(:password=).and_return(password)
+      volume.encryption_password = password
+    end
+
+    context "if volume has not encryption password" do
+      let(:password) { nil }
+
+      it "returns the plain device" do
+        device = encrypter.device_for(volume, plain_device)
+        expect(device).to eq(plain_device)
+      end
```

Comment:
Maybe it would also be a good idea to verify that `create_encryption` is not called at all if password is nil